### ISSUE TITLE
Bootstrap broken on macOS, requires `zsh`

### ISSUE THREAD

**Issue by fabianhjr** [idris-lang/Idris2#1612]
While packaging 0.4.0 for nixOS the following issue came up:

```
bootstrap/idris2-rktboot.sh
1-#!/bin/sh
2-
3-set -e # exit on any error
4-
5-if [ "$(uname)" = Darwin ]; then
6:    DIR=$(zsh -c 'printf %s "$0:A:h"' "$0")
7-else
8-    DIR=$(dirname "$(readlink -f -- "$0")")
9-fi

bootstrap/idris2-boot.sh
10-if [ "$(uname)" = Darwin ]; then
11:    DIR=$(zsh -c 'printf %s "$0:A:h"' "$0")
12-else
13-    DIR=$(dirname "$(readlink -f -- "$0")")
14-fi
```

From:
```
src/Compiler/Scheme/Racket.idr
331-startRacket : String -> String -> String -> String
332-startRacket racket appdir target = unlines
333-    [ "#!/bin/sh"
334-    , ""
335-    , "set -e # exit on any error"
336-    , ""
337-    , "if [ \"$(uname)\" = Darwin ]; then"
338:    , "  DIR=$(zsh -c 'printf %s \"$0:A:h\"' \"$0\")"
339-    , "else"
340-    , "  DIR=$(dirname \"$(readlink -f -- \"$0\")\")"
341-    , "fi"

src/Compiler/Scheme/Chez.idr
377-startChezPreamble : String
378-startChezPreamble = unlines
379-    [ "#!/bin/sh"
380-    , ""
381-    , "set -e # exit on any error"
382-    , ""
383-    , "if [ \"$(uname)\" = Darwin ]; then"
384:    , "  DIR=$(zsh -c 'printf %s \"$0:A:h\"' \"$0\")"
385-    , "else"
386-    , "  DIR=$(dirname \"$(readlink -f -- \"$0\")\")"
387-    , "fi"
388-    , ""  -- so that the preamble ends with a newline
```

**Comment by fabianhjr** [comment]
From the commit: https://github.com/idris-lang/Idris2/commit/b457d15701f6f359d66c6e32af29561b44bfd81e

```
 [ fix #221 ] realpath alternative for macOS

Zsh is available by default on macOS
readlink -f is commonly supported on Linux and BSD
```

**Comment by wchresta** [comment]
I believe this was changed with https://github.com/idris-lang/Idris2/pull/1418
